Solution1/ClassLibrary1/NumericUpDownLight.cs

```diff
@@ -0,0 +1,249 @@
+        private EventHandler _valueChanged = null;
+        private HorizontalAlignment _textAlign = HorizontalAlignment.Left;
+        private Color _mouseHoverBackColor = Color.Blue;
+
+        private readonly CustomTextBox TextBox = new CustomTextBox();
+        private readonly CustomButton ButtonUp = new CustomButton();
+        private readonly CustomButton ButtonDown = new CustomButton();
+
+        private bool IsBackgroundChanged = false;
+
+        [Browsable(false)]
+        public override string Text { get; set; }
+
+        [
+        Category("Data"),
+        Description("Specifies the value of the control.")
+        ]
+        public int Value
+        {
+            get
+            {
+                return _value;
+            }
+            set
+            {
+                _value = value;
+                base.Invalidate();
+            }
+        }
+
+        [
+        Category("Property Changed"),
+        Description("ValueChanged event triggers after the value was changed")
+        ]
+        public event EventHandler ValueChanged
+        {
+            add
+            {
+                _valueChanged = value;
+            }
+            remove
+            {
+                _valueChanged = null;
+            }
+        }
+
+        [
+        Category("Appearance"),
+        Description("Specifies the alignment of the text")
+        ]
+        public HorizontalAlignment TextAlign
+        {
+            get
+            {
+                return _textAlign;
+            }
+            set
+            {
+                _textAlign = value;
+                base.Invalidate();
+            }
+        }
+
+        [
+        Category("Appearance"),
+        Description("Specifies the mouse hover back color of the buttons")
+        ]
+        public Color MouseHoverBackColor
+        {
+            get
+            {
+                return _mouseHoverBackColor;
+            }
+            set
+            {
+                _mouseHoverBackColor = value;
+            }
+        }
+
+        public NumericUpDownLight()
+        {
+            DoubleBuffered = true;
+
+            TextBox = new CustomTextBox
+            {
+                Text = _value.ToString(),
+                TextAlign = TextAlign,
+            };
+
+            ButtonUp = new CustomButton
+            {
+                Text = "▲",
+                OnMouseClick = this.IncrementValue
+            };
+
+            ButtonDown = new CustomButton
+            {
+                Text = "▼",
+                OnMouseClick = this.DecrementValue
+            };
+        }
+
+        protected override void OnMouseClick(MouseEventArgs e)
+        {
+            base.OnMouseClick(e);
+
+            var upButtonRectangle = GetRectangleForButtonUp();
+            var downButtonRectangle = GetRectangleForButtonDown();
+
+            if (upButtonRectangle.Contains(e.Location))
+            {
+                ButtonUp.OnMouseClick(this, e);
```

Comment:
Please think more about this. I don't see any place in CustomButton which needs this call. And CustomButton class is not public, so where it is used in this library?
In ideal case it might look like this:
- CustomButton handles catches mouse events and fires OnClick event
- NumericUpDownLight subscribes on button events and does some actions in event handlers
- in that case CustomButton will be able to work without NumericUpdown and be used in other controls
If CustomButton is more like a light-weight model and can't handle mouse events, then having OnMouseClick in it is useless. If I'm not wrong, it does nothing now and can be removed